File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def isHappy(self, n: int) -> bool:
        vis = set()
        def check(v):
            if v == 1:
                return True

            if v in vis:
                return False

            vis.add(v)

            t = 0
            while v:
                t += (v % 10)**2
                v //= 10
            
            return check(t)
        
        return check(n)

        

File: Data Structures & Algorithms/valid-palindrome-ii/submission-0.cs
public class Solution {
    public bool ValidPalindrome(string s) {

        bool Check(int lo, int hi, int k)
        {
            while (lo < hi)
            {
                if (s[lo] != s[hi] )
                {
                    return k > 0 ? (
                        Check(lo + 1, hi, k - 1) ||
                        Check(lo, hi - 1, k - 1)
                    ): false;
                }
                lo += 1;
                hi -= 1;
            }

            return true;
        }

        var size = s.Length;
        return Check(0, size - 1, 1);
    }
}